### PR TITLE
Implement mocking SocialGet

### PR DESCRIPTION
**Pull request by sekaiking**
It's supposed to work by taking data json from bos loader server and replace it when the user calls Social.get. Didn't test it extensively, so it may be buggy and inconsistent with Social.get from SocialDB

Works with this: https://github.com/sekaiking/bos-workspace/pull/9

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -57,6 +61,54 @@ function App(props) {
 
   const location = window.location;
 
+  const cc = useCache();
+  cc.socialGet = (near, keys, recursive, blockId, options, invalidate) => {
```
Comment (elliotBraem) [inline src/App.js:65]: This is awesome. We could even move this into the VM

**Comment by elliotBraem** [comment]
Unless you have any reason not to, I'm down to merge to main. It's helping my development

**Comment by elliotBraem** [comment]
@sekaiking I had to revert this PR; once deployed, the Social.get takes longer in each of the widgets, and so a wave of red boxes overtakes the screen until it's able to resolve. 

I think this is only happening because we are mocking the Social.get via the gateway -- this problem should go away if we implement it directly into the VM. 

We can pass dataMap into the VM and then access it via this.vm.widgetConfigs.dataMap

**Comment by sekaiking** [comment]
> @sekaiking I had to revert this PR; once deployed, the Social.get takes longer in each of the widgets, and so a wave of red boxes overtakes the screen until it's able to resolve.
> 
> I think this is only happening because we are mocking the Social.get via the gateway -- this problem should go away if we implement it directly into the VM.
> 
> We can pass dataMap into the VM and then access it via this.vm.widgetConfigs.dataMap

I see. You are right It's better if we implement it directly under the VM like redirectMap. I will try to study the VM code more later.

**Pull request merged**